app/models/vacation.rb

```diff
@@ -29,11 +29,16 @@ def validates_start_date_less_than_end_date
   def validates_work_time
     return unless user
 
-    any_work_time = WorkTime.where('((starts_at::timestamp::date >= :start_date AND starts_at::timestamp::date <= :end_date) OR
-                                    (ends_at::timestamp::date >= :start_date AND ends_at::timestamp::date <= :end_date) OR
-                                    ((starts_at::timestamp::date, starts_at::timestamp::date) OVERLAPS (:start_date, :end_date))) AND
-                                    discarded_at IS NULL AND user_id = :user_id', start_date: start_date, end_date: end_date, user_id: user_id).any?
-    errors.add(:base, :work_time_exists) if any_work_time
+    work_times = WorkTime.where('((starts_at >= :start_date AND starts_at <= :end_date) OR
+                                    (ends_at >= :start_date AND ends_at <= :end_date) OR
+                                    ((starts_at, starts_at) OVERLAPS (:start_date, :end_date))) AND
+                                    discarded_at IS NULL AND user_id = :user_id',
+                                start_date: start_date.beginning_of_day, end_date: end_date.end_of_day, user_id: user_id)
+    return if work_times.blank?
+
+    vacation_ids = work_times.pluck(:vacation_id).uniq
+    errors.add(:base, :vacation_exists) if vacation_ids.any? { |x| x.is_a?(Numeric) }
```

Comment:
to simplify this i would use following:

if work_times.any?(&:vacation_id)
  errors.add(:base, :vacation_exists)
else
  errors.add(:base, :work_time_exists)
end